### PR TITLE
Pathlib /unit/utils

### PR DESCRIPTION
**Pull request by Lembrun**
#176 , haven't touched tests/unit/utils/test_urlutils as discussed in IRC

### CODE REVIEW

**Comment by The-Compiler** [comment]
Looks like stuff [started failing](https://github.com/qutebrowser/qutebrowser/pull/6291/checks?check_run_id=2125739784) on Windows - I haven't taken a closer look yet, but it looks like some parts fail due to expecting `C:\Users\RUNNER~1\AppData\...` but getting `C:\Users\runneradmin\AppData\...`.

Gotta love Windows... I'll take a closer look once I feel like dealing with it.

File: qutebrowser/utils/standarddir.py [hunk 1]
```diff
@@ -301,7 +301,7 @@ def _from_args(
         suffix = basedir_suffix[typ]
     except KeyError:  # pragma: no cover
         return None
-    return os.path.abspath(os.path.join(args.basedir, suffix))
```
Comment (Lembrun) [inline qutebrowser/utils/standarddir.py:304]: I've located the problem here, apparently pathlib.resolve doesn't behave the same way on windows for non existing path [https://stackoverflow.com/questions/42513056/how-to-get-absolute-path-of-a-pathlib-path-object](url) (though I haven't checked in a vm)

I'm going to eat but there are still some changes that I need to push for this one

**Review by The-Compiler — CHANGES_REQUESTED** [review]
I don't think the changes in `qutebrowser/utils/standarddir.py` make much sense (and they are also separate from the changes in the tests).

For `standarddir`, I'd rather switch over the `standarddir` functions entirely, rather than sprinkling `standarddir` and `str` conversions everywhere. But that will need some more planning (since it affects the entire codebase pretty much), so it really shouldn't be part of this PR I think.

**Comment by The-Compiler** [comment]
Thanks! Merged with some more simplifications:

- 816ec60fe Remove changes to test_standarddir
- 27ceb8196 Simplify paths in test_jinja, _qtutils and _version

**Pull request merged**